# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmFrameIter.h
# Commit: f981949ae991
# Full Hash: f981949ae991960c2099ea85435ddbb660a4229c
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-31 21:05:17
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 3: Implement the algorithm for obtaining tls and use it for wasm signal handling. r=lth
#   
#   This is the third part of series of patches to Frame without tls pointer.
#   Here we preserve initial tls in all entry stubs and then use it to find a proper tls instance for a given frame.
#   
# ==============================================================================

diff -r 6d85470024c3 -r f981949ae991 js/src/wasm/WasmFrameIter.h
--- a/js/src/wasm/WasmFrameIter.h	Sat Oct 31 09:56:05 2020 +0000
+++ b/js/src/wasm/WasmFrameIter.h	Sat Oct 31 09:56:05 2020 +0000
@@ -233,6 +233,10 @@
 void GenerateFunctionEpilogue(jit::MacroAssembler& masm, unsigned framePushed,
                               FuncOffsets* offsets);
 
+// Iterates through frames for either possible cross-instance call or an entry
+// stub to obtain tls that corresponds to the passed fp.
+TlsData* GetNearestEffectiveTls(Frame* fp);
+
 // Describes register state and associated code at a given call frame.
 
 struct UnwindState {